### DISCUSSION TITLE
Free display arrangement instead grid

### DISCUSSION THREAD

**Discussion by davidkennedydev** [debauchee/barrier#1451] [Ideas]
Today I'm using 2 displays and another external computer that I was connecting using Barrier and it's amazing!
However now I had to change my setup arrangement and start to get issues with the grid display arrangement, at Gnome, there is a way to arrange displays freely:
![image](https://user-images.githubusercontent.com/6628028/143453235-64614de7-b460-422a-a3bb-c848df6814b8.png)

Barrier just has a grid for arrangement and seems that they don't take into account second displays the grid just look to the host machine as a single display computer:
![image](https://user-images.githubusercontent.com/6628028/143453491-ae5462d4-7341-4956-a36e-0225e1619e1e.png)

There is a way to have a similar free display configuration instead of a grid and take into account multiple displays at one computer?

**Comment by davidkennedydev** [comment]
# Looking for changes

## GUI

It's a grid UI element but could be a free drag and drop element
https://github.com/debauchee/barrier/blob/fc045fc79326cef966405b1cc578e8f062ae5294/src/gui/src/ServerConfig.cpp#L29-L41

## Server

IScreen seems already enough abstract because it has as parameters a left-top first-pixel position, width, and height:
https://github.com/debauchee/barrier/blob/fc045fc79326cef966405b1cc578e8f062ae5294/src/lib/server/BaseClientProxy.h#L64-L65

The server implementation seems to use these parameters generically:
https://github.com/debauchee/barrier/blob/fc045fc79326cef966405b1cc578e8f062ae5294/src/lib/server/Server.cpp#L450-L461

## TL;DR

Maybe the main blockers existing today are the rigid GUI and the server's inability to treat multiple displays in one same machine.

**"The inability to treat multiple displays on the same machine"** seems more complex to fix but could be done in a separate effort.

**Comment by davidkennedydev** [comment]
https://github.com/debauchee/barrier/issues/1505

**Comment by davidkennedydev** [comment]
https://github.com/debauchee/barrier/issues/1112

**Comment by davidkennedydev** [comment]
https://github.com/debauchee/barrier/issues/233

**Comment by davidkennedydev** [comment]
https://github.com/debauchee/barrier/issues/127
